pkg/synchronizer/githubsynchronizer/githubsynchronizer.go

```diff
@@ -158,19 +158,32 @@ func (sync *ChannelSynchronizer) processYamlFile(chn *chv1.Channel, files []os.F
 			if strings.EqualFold(filepath.Ext(f.Name()), ".yml") || strings.EqualFold(filepath.Ext(f.Name()), ".yaml") {
 				// check it it is Kubernetes resource
 				klog.V(debugLevel).Info("scanning file ", f.Name())
-				file, _ := ioutil.ReadFile(filepath.Join(filepath.Clean(cloneDir), filepath.Clean(dir), filepath.Clean(f.Name())))
-				t := kubeResource{}
+				file, err := ioutil.ReadFile(filepath.Join(filepath.Clean(cloneDir), filepath.Clean(dir), filepath.Clean(f.Name())))
 
-				err := yaml.Unmarshal(file, &t)
 				if err != nil {
-					klog.Info("Failed to unmarshal Kubernetes resource, err:", err)
-					break
+					klog.Error(err, "Failed to read YAML file "+filepath.Join(filepath.Clean(cloneDir), filepath.Clean(dir), filepath.Clean(f.Name())))
+					continue
 				}
 
-				if t.APIVersion == "" || t.Kind == "" {
-					klog.V(debugLevel).Info("Not a Kubernetes resource")
-				} else if err := sync.handleSingleDeployable(chn, dir, f, file, t, newDplList); err != nil {
-					klog.Errorf("Failed to handle single deployable, err: %+v", err)
+				resources := utils.ParseKubeResoures(file)
+
+				if len(resources) == 0 {
+					klog.Info("The file contains no Kubernetes resource.")
+					continue
+				} else {
```

Comment:
I feel we can drop this `else`, since the `if` has a `continue` statement.

Reply:
Thanks for pointing that out. In this case, `continue` statement is useless so I am removing that.